server/src/main/java/org/elasticsearch/script/field/WriteField.java

```diff
@@ -0,0 +1,386 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.script.field;
+
+import org.elasticsearch.common.util.Maps;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+
+public class WriteField implements Field<Object> {
+    protected String path;
+    protected final Supplier<Map<String, Object>> rootSupplier;
```

Comment:
What is the advantage of having this as a supplier?

Reply:
The root may change because it was completely replaced by the user.

Reply:
Makes sense, thanks.